streams/shared/src/main/scala/scalaz/stream/ZSink.scala

```diff
@@ -517,6 +551,12 @@ object ZSink {
       def extractPure(state: State): Either[E, Nothing]        = Left(e)
     }
 
+  /**
+   * Creates a sink by folding over a structure of type `S`.
+   *
+   * TODO Hum, should this one be called `unfold` instead? `Step` is a bit like `Option`
```

Comment:
`Step` is indeed like `Option` but I'd say it's more of a short-circuiting fold than an unfold.

Reply:
Got it 👍